src/main/scala/fi/vm/sade/hakurekisteri/integration/ytl/YtlIntegration.scala

```diff
@@ -141,56 +141,70 @@ class YtlIntegration(
     persons: Set[HetuPersonOid],
     failureEmailSender: FailureEmailSender
   ): Unit = {
-    val personOidToHetu: Map[String, String] =
-      persons.map(person => person.personOid -> person.hetu).toMap
-
-    logger.info(s"Get possible additional hetus for ${persons.size} persons")
-    val onrPersons =
-      Await.result(oppijaNumeroRekisteri.getByOids(persons.map(_.personOid)), 60.seconds)
-    val hetuToAllHetus =
-      onrPersons.map(person => personOidToHetu(person._1) -> person._2.kaikkiHetut)
-
-    // Now that we query with possible previous hetus as well, we also have to match response data with them.
-    val allHetusToPersonOids: Map[String, String] = persons
-      .flatMap(person => {
-        val hetus = hetuToAllHetus.getOrElse(person.hetu, Some(List(person.hetu))) match {
-          case Some(h) => h
-          case None    => List(person.hetu)
-        }
-        hetus.map(hetu => hetu -> person.personOid)
-      })
-      .toMap
-
-    val personsGrouped: Iterator[Set[HetuPersonOid]] = persons.grouped(10000)
 
-    logger.info(s"Handle hakemukset for ${persons.size} persons")
-    val personOidsWithAliases = personsGrouped
-      .map(ps =>
-        Await.result(
-          oppijaNumeroRekisteri.enrichWithAliases(ps.map(_.personOid)),
-          Duration(5, TimeUnit.MINUTES)
+    try {
+      logger.info(s"About to fetch possible additional hetus for ${persons.size} persons")
+      val personOidToHetu: Map[String, String] =
+        persons.map(person => person.personOid -> person.hetu).toMap
+
+      val futureOnrPersons = oppijaNumeroRekisteri.getByOids(persons.map(_.personOid))
+
+      val futureHetuToAllHetus =
+        futureOnrPersons.map(_.map(person => personOidToHetu(person._1) -> person._2.kaikkiHetut))
+
+      // Now that we query with previous hetus as well, we also have to have a way to match response data with them.
+      val futureHetusToPersonOids = Future
+        .sequence(
+          persons
+            .map(person => {
+              val futureHetus =
+                futureHetuToAllHetus.map(_.getOrElse(person.hetu, Some(List(person.hetu))) match {
+                  case Some(h) => h
+                  case None    => List(person.hetu)
+                })
+              futureHetus.map(_.map(hetu => hetu -> person.personOid).toMap)
+            })
         )
-      )
-      .reduce((a, b) =>
-        PersonOidsWithAliases(
-          a.henkiloOids ++ b.henkiloOids,
-          a.aliasesByPersonOids ++ b.aliasesByPersonOids
+        .map(_.flatten.toMap)
```

Comment:
Menisiköhän ilman sequencea kivasti näin:
`      val futureHetusToPersonOids: Future[Map[String, String]] =
        futureHetuToAllHetus.map(futureHetuResult =>
          persons
            .flatMap(person => {
              val hetut = futureHetuResult.getOrElse(person.hetu, Some(List(person.hetu))) match {
                case Some(h) => h
                case None    => List(person.hetu)
              }
              hetut.map(hetu => hetu -> person.personOid)
            })
            .toMap
        )
`

Reply:
Menisi!